tests/src/test/java/io/streamnative/pulsar/handlers/kop/DifferentNamespaceTest.java

```diff
@@ -0,0 +1,117 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.pulsar.common.policies.data.TenantInfo;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * test produce/consume in different namespaces.
+ */
+@Slf4j
+public class DifferentNamespaceTest extends KopProtocolHandlerTestBase {
+
+    private static final String DEFAULT_TENANT = "public";
+    private static final String DEFAULT_NAMESPACE = "default";
+    private static final String ANOTHER_TENANT = "my-tenant";
+    private static final String ANOTHER_NAMESPACE = "my-ns";
+
+
+    @DataProvider(name = "topics")
+    public static Object[][] topics() {
+        return new Object[][] {
+                { "topic-0" },
+                { DEFAULT_TENANT + "/" + DEFAULT_NAMESPACE + "/" + "topic-1" },
+                { "persistent://" + DEFAULT_TENANT + "/" + DEFAULT_NAMESPACE + "/" + "topic-2" },
+                { ANOTHER_TENANT + "/" + ANOTHER_NAMESPACE + "/" + "topic-3" },
+                { "persistent://" + ANOTHER_TENANT + "/" + ANOTHER_NAMESPACE + "/" + "topic-4" },
+        };
+    }
+
+    @BeforeClass
+    @Override
+    protected void setup() throws Exception {
+        super.internalSetup();
+
+        admin.tenants().createTenant(ANOTHER_TENANT,
+                new TenantInfo(Sets.newHashSet("admin_user"), Sets.newHashSet(super.configClusterName)));
+        admin.namespaces().createNamespace(ANOTHER_TENANT + "/" + ANOTHER_NAMESPACE);
+    }
+
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    @Test(timeOut = 20000, dataProvider = "topics")
+    void testSimpleProduceAndConsume(String topic) {
+        final int numMessages = 10;
+        final String messagePrefix = topic + "-msg-";
+
+        @Cleanup
+        KProducer kProducer = new KProducer(topic, false, getKafkaBrokerPort());
+        for (int i = 0; i < numMessages; i++) {
+            final int key = i;
+            kProducer.getProducer().send(new ProducerRecord<>(topic, key, messagePrefix + key),
+                    (recordMetadata, e) -> {
+                if (e == null) {
+                    log.info("Successfully send {} to {}-partition-{}",
+                            key, recordMetadata.topic(), recordMetadata.partition());
+                } else {
+                    log.error("Failed to send {}", key);
+                    fail("Failed to send " + key);
+                }
+            });
+        }
+
+        @Cleanup
+        KConsumer kConsumer = new KConsumer(topic, getKafkaBrokerPort(), true);
+        kConsumer.getConsumer().subscribe(Collections.singleton(topic));
```

Comment:
Would you please also add some test related with `consumer.commit()`? so that offsetAcker could be tested.
especially, to check whether the offsetTopic is created as it is config-ed?
```
        OffsetConfig offsetConfig = OffsetConfig.builder()
            .offsetsTopicName(kafkaConfig.getKafkaMetadataTenant() + "/"
                + kafkaConfig.getKafkaMetadataNamespace()
                + "/" + Topic.GROUP_METADATA_TOPIC_NAME)
```

Reply:
OK, I'll add the test